### PR TITLE
Add ability to get job artifact by Ref

### PR DESCRIPTION
**Pull request by shivan**
see #563

### CODE REVIEW

File: NGitLab/Models/JobArtifactQuery.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    public class JobArtifactQuery
+    {
+        [JsonPropertyName("ref_name")]
+        public string RefName;
+
+        [JsonPropertyName("artifact_path")]
+        public string ArtifactPath;
+
+        [JsonPropertyName("job")]
+        public string JobName;
```
Comment (louis-z) [inline NGitLab/Models/JobArtifactQuery.cs]: Please define properties rather than fields.

**Review by louis-z — CHANGES_REQUESTED** [review]

File: NGitLab/Models/JobArtifactQuery.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Text.Json.Serialization;
+
+namespace NGitLab.Models
+{
+    public class JobArtifactQuery
```
Comment (louis-z) [inline NGitLab/Models/JobArtifactQuery.cs]:
```suggestion
    public sealed class JobArtifactQuery
```

File: NGitLab/Models/JobArtifactQuery.cs (hunk 1 above)
Comment (shivan) [reply NGitLab/Models/JobArtifactQuery.cs]: ok

**Comment by shivan** [comment]
@louis-z any further changes requested? Or all ok now?

**Comment by louis-z** [comment]
> @louis-z any further changes requested? Or all ok now?

I had a quick look and it seems fine. Can you merge or rebase from the main branch? Note that there are conflicts. I'll have one more look at it tomorrow.

**Pull request merged**